lib/core/utils/get-flattened-tree.js

```diff
@@ -38,6 +39,20 @@ function getSlotChildren(node) {
   return retVal;
 }
 
+/**
+ * Create a virtual node
+ * @param {Node} node the current node
+ * @param {VirtualNode} parent the parent VirtualNode
+ * @param {String} shadowId, optional ID of the shadow DOM that is the closest shadow ancestor of the node
+ * @return {VirtualNode}
+ */
+function createNode(node, parent, shadowId) {
+  const vNode = new VirtualNode(node, parent, shadowId);
+  cacheNodeSelectors(vNode, cache.get('selectorMap'));
```

Comment:
It messed up a few tests when having the cache of the node set and when tests would create a VirtualNode that wasn't part of the tree itself. I think it's better that this only cache the selector when in the flatten tree script since it's specifically caching the information about the tree (and not one off vNodes).